iree/hal/cts/command_buffer_dispatch_test.h

```diff
@@ -0,0 +1,143 @@
+
+namespace iree {
+namespace hal {
+namespace cts {
+
+class command_buffer_dispatch_test : public CtsTestBase {
+ protected:
+  void PrepareAbsExecutable() {
+    IREE_ASSERT_OK(iree_hal_executable_cache_create(
+        device_, iree_make_cstring_view("default"), &executable_cache_));
+
+    iree_hal_descriptor_set_layout_binding_t descriptor_set_layout_bindings[] =
+        {
+            {0, IREE_HAL_DESCRIPTOR_TYPE_STORAGE_BUFFER},
+            {1, IREE_HAL_DESCRIPTOR_TYPE_STORAGE_BUFFER},
+        };
+    IREE_ASSERT_OK(iree_hal_descriptor_set_layout_create(
+        device_, IREE_HAL_DESCRIPTOR_SET_LAYOUT_USAGE_TYPE_PUSH_ONLY,
+        IREE_ARRAYSIZE(descriptor_set_layout_bindings),
+        descriptor_set_layout_bindings, &descriptor_set_layout_));
+    IREE_ASSERT_OK(iree_hal_executable_layout_create(
+        device_, /*push_constants=*/0, /*set_layout_count=*/1,
+        &descriptor_set_layout_, &executable_layout_));
+
+    iree_hal_executable_spec_t executable_spec;
+    executable_spec.caching_mode =
+        IREE_HAL_EXECUTABLE_CACHING_MODE_ALIAS_PROVIDED_DATA;
+    executable_spec.executable_format =
+        iree_make_cstring_view(get_test_executable_format());
+    executable_spec.executable_data = get_test_executable_data(
+        iree_make_cstring_view("command_buffer_dispatch_test.bin"));
+    executable_spec.executable_layout_count = 1;
+    executable_spec.executable_layouts = &executable_layout_;
+
+    IREE_ASSERT_OK(iree_hal_executable_cache_prepare_executable(
+        executable_cache_, &executable_spec, &executable_));
+  }
+
+  void CleanupExecutable() {
+    iree_hal_executable_release(executable_);
+    iree_hal_executable_layout_release(executable_layout_);
+    iree_hal_descriptor_set_layout_release(descriptor_set_layout_);
+    iree_hal_executable_cache_release(executable_cache_);
+  }
+
+  iree_hal_executable_cache_t* executable_cache_ = NULL;
+  iree_hal_descriptor_set_layout_t* descriptor_set_layout_ = NULL;
+  iree_hal_executable_layout_t* executable_layout_ = NULL;
+  iree_hal_executable_t* executable_ = NULL;
+};
+
+TEST_P(command_buffer_dispatch_test, DispatchAbs) {
+  PrepareAbsExecutable();
+
+  iree_hal_command_buffer_t* command_buffer;
+  IREE_ASSERT_OK(iree_hal_command_buffer_create(
+      device_,
+      IREE_HAL_COMMAND_BUFFER_MODE_ONE_SHOT |
+          IREE_HAL_COMMAND_BUFFER_MODE_ALLOW_INLINE_EXECUTION,
+      IREE_HAL_COMMAND_CATEGORY_DISPATCH, IREE_HAL_QUEUE_AFFINITY_ANY,
+      &command_buffer));
+
+  IREE_ASSERT_OK(iree_hal_command_buffer_begin(command_buffer));
+
+  // Create input and output buffers.
+  iree_hal_buffer_view_t* input_buffer_view = NULL;
+  float input_data[1] = {-2.5f};
+  IREE_ASSERT_OK(iree_hal_buffer_view_allocate_buffer(
+      device_allocator_, /*shape=*/NULL,
+      /*shape_rank=*/0, IREE_HAL_ELEMENT_TYPE_FLOAT_32,
+      IREE_HAL_ENCODING_TYPE_DENSE_ROW_MAJOR,
+      IREE_HAL_MEMORY_TYPE_HOST_LOCAL | IREE_HAL_MEMORY_TYPE_DEVICE_VISIBLE,
```

Comment:
host local/device visible buffers are likely going to stop working for dispatches at some point soon (needed for WebGPU, which can't do that) - if these can be pure DEVICE_LOCAL that'd be better (and you don't need this to be host local anyway)